core/src/main/scala/org/bitcoins/core/script/interpreter/ScriptInterpreter.scala

```diff
@@ -422,397 +420,448 @@ sealed abstract class ScriptInterpreter extends BitcoinSLogger {
     }
   }
 
+  private def executeProgram(
+      program: PreExecutionScriptProgram): ExecutedScriptProgram = {
+    logger.trace("Stack: " + program.stack)
+    logger.trace("Script: " + program.script)
+    val scriptByteVector = BitcoinSUtil.toByteVector(program.script)
+    if (scriptByteVector.length > 10000) {
+      logger.error("We cannot run a script that is larger than 10,000 bytes")
+      ScriptProgram(ScriptProgram.toExecutionInProgress(program),
+                    ScriptErrorScriptSize)
+    } else {
+      loop(ScriptProgram.toExecutionInProgress(program, Some(program.stack)), 0)
+    }
+  }
+
+  @tailrec
+  private def completeProgramExecution(
+      program: ExecutedScriptProgram,
+      opCount: Int): ExecutedScriptProgram = {
+    val countedOps = program.originalScript
+      .count(BitcoinScriptUtil.countsTowardsScriptOpLimit)
+    logger.trace("Counted ops: " + countedOps)
+
+    if (countedOps > MAX_SCRIPT_OPS && program.error.isEmpty) {
+      completeProgramExecution(ScriptProgram(program, ScriptErrorOpCount),
+                               opCount)
+    } else {
+      program
+    }
+  }
+
   /**
     * The execution loop for a script
     *
     * @param program the program whose script needs to be evaluated
     * @return program the final state of the program after being evaluated by the interpreter
     */
-  @tailrec
+  //@tailrec
```

Comment:
if this can't be re-added, i think this PR is a regression in terms of performance. We should definitely take out stuff that forces this not to be tail recursive.

Reply:
This is outdated, it is already back